packages/server/src/service/org/passport.ts

```diff
@@ -243,12 +244,23 @@ export const registerPassportEndpoints = (app: Express.Application) => {
     async (req, res) => {
       const { oid } = req.params
       const uid = getUid(req)
-      const stateInfos = await firestoreService.fetchRegistrations(uid, oid) || []
+      const isDownloading = csvMailingQueue.isDownloading(uid)
+      const { name, emails } = await firestoreService.getOrgUser(uid)
+      if (!isDownloading) {
+        req.flash('success', `We're sending all the registrations in an attachment to your email (${emails[0]})`)
+      } else {
+        req.flash('success', `There seems to be already a request to send the registrations to your email, if you didn't receive a copy of the file please wait a few minutes and you will be able to make this request again.`)
+      }
+      res.redirect('/dashboard')
 
-      const csvString = toCSVSting(stateInfos)
-      res.contentType('text/csv')
-      res.setHeader('Content-Disposition', `attachment; filename=${oid}-data.csv`)
-      res.send(csvString)
+      // Mail the file after handling the request response, so don't have
+      // to await for long promises.
+      if (!isDownloading) {
```

Comment:
I don't totally understand this code and am not being as thorough as I could be, but if this isn't happening already, can we only send the email if the CSV file is large and let the download happen if the CSV file is small enough?

Feel free to disagree.  This could be a bad 'product' decision.

Reply:
I thought of doing that, but to know if the file is large we would have to execute the query since we don't have a easy way to get the number of items in the collection.

Reply:
Can you add something below the 'Download All' button that indicates we'll email it, then?